### PR TITLE
LPD-35470 Move SCIM Poshi tests to integration tests

### PR DESCRIPTION
**Pull request by manuelecastro**
Related issue: [LPD-35470](https://liferay.atlassian.net/browse/LPD-35470)

These changes are moving the following tests from functional poshi tests to integration tests:

- [SCIM#SCIMClientCannotPostAUserIfTheyAreLinkedToAnotherClient](https://testray.liferay.com/#/project/35392/cases/92489) - is already being satisfied by: https://github.com/liferay-appsec/liferay-portal/blob/0ba14ecc71f771e834732cad23954ee8e69d8521/modules/dxp/apps/scim/scim-rest-test/src/testIntegration/java/com/liferay/scim/rest/resource/v1_0/test/UserResourceTest.java#L267

- [SCIM#SCIMClientCannotUpdateAUserOfADifferentClient](https://testray.liferay.com/#/project/35392/cases/96496) - is now satisfied by: https://github.com/liferay-appsec/liferay-portal/blob/0ba14ecc71f771e834732cad23954ee8e69d8521/modules/dxp/apps/scim/scim-rest-test/src/testIntegration/java/com/liferay/scim/rest/resource/v1_0/test/UserResourceTest.java#L307

### CODE REVIEW

**Comment by manuelecastro** [comment]
ci:test:sf

**Comment by manuelecastro** [comment]
ci:test:scim

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

File: modules/dxp/apps/scim/scim-rest-test/src/testIntegration/java/com/liferay/scim/rest/resource/v1_0/test/UserResourceTest.java [hunk 1]
```diff
@@ -304,21 +304,34 @@ public void testPutV2User() throws Exception {
...
+		ScimTestUtil.saveSCIMClientId(
+			com.liferay.portal.kernel.model.User.class.getName(),
+			GetterUtil.getLong(user2.getId()), TestPropsValues.getCompanyId(),
+			ScimClientUtil.generateScimClientId(
+				"scim-client-test" + RandomTestUtil.randomString()));
+
+		assertHttpResponseStatusCode(
+			409, userResource.putV2UserHttpResponse(user2.getId(), user2));
+
+		User user3 = testDeleteV2User_addUser();
```
Comment (stian-sigvartsen) [inline modules/dxp/apps/scim/scim-rest-test/src/testIntegration/java/com/liferay/scim/rest/resource/v1_0/test/UserResourceTest.java:320]: I think we avoid creating a 3rd user here if you assert the successful update prior to changing the `scimClientId` of user2?

**Review by stian-sigvartsen — CHANGES_REQUESTED** [review]
Hi @manuelecastro . This is a beautifully presented PR! 
Please consider my comment about the optimization.

**Comment by manuelecastro** [comment]
Hey @stian-sigvartsen ! I've Just pushed some changes doing what you suggested. :)

**Comment by manuelecastro** [comment]
ci:test:scim

**Comment by manuelecastro** [comment]
All related tests have passed! :heavy_check_mark: 

image

**Review by stian-sigvartsen — APPROVED** [review]
Looks great @manuelecastro ! Thanks.

**Comment by stian-sigvartsen** [comment]
ci:forward

**Comment by stian-sigvartsen** [comment]
ci:forward

**Comment by stian-sigvartsen** [comment]
ci:forward:force

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
ci:forward

**Pull request closed without merging**